Kernel/Storage/NVMe/NVMeDefinitions.h

```diff
@@ -160,3 +160,71 @@ struct NVMeSubmission {
     LittleEndian<u32> cdw14;
     LittleEndian<u32> cdw15;
 };
+
+struct NVMeRWCmd {
+    u8 op;
+    u8 flags;
+    LittleEndian<u16> cmdid;
+    LittleEndian<u32> nsid;
+    LittleEndian<u64> rsvd;
+    LittleEndian<u64> metadata;
+    struct DataPtr data_ptr;
+    LittleEndian<u64> slba;
+    LittleEndian<u16> length;
+    LittleEndian<u16> control;
+    LittleEndian<u32> dsmgmt;
+    LittleEndian<u32> reftag;
+    LittleEndian<u16> apptag;
+    LittleEndian<u16> appmask;
+};
+
+struct NVMeIdentifyCmd {
+    u8 op;
+    u8 flags;
+    LittleEndian<u16> cmdid;
+    LittleEndian<u32> nsid;
+    LittleEndian<u64> rsvd1[2];
+    struct DataPtr data_ptr;
+    u8 cns;
+    u8 rsvd2;
+    LittleEndian<u16> ctrlid;
+    u8 rsvd3[3];
+    u8 csi;
+    u64 rsvd4[2];
+};
+
+struct NVMeCreateCQCmd {
+    u8 op;
+    u8 flags;
+    LittleEndian<u16> cmdid;
+    u32 rsvd1[5];
+    LittleEndian<u64> prp1;
+    u64 rsvd2;
+    LittleEndian<u16> cqid;
+    LittleEndian<u16> qsize;
+    LittleEndian<u16> cq_flags;
+    LittleEndian<u16> irq_vector;
+    u64 rsvd12[2];
+};
+
+struct NVMeCreateSQCmd {
+    u8 op;
+    u8 flags;
+    LittleEndian<u16> cmdid;
+    u32 rsvd1[5];
+    LittleEndian<u64> prp1;
+    u64 rsvd2;
+    LittleEndian<u16> sqid;
+    LittleEndian<u16> qsize;
+    LittleEndian<u16> sq_flags;
+    LittleEndian<u16> cqid;
+    u64 rsvd12[2];
+};
+
+union NVMeSubmission {
+    NVMeGenericCmd generic;
+    NVMeIdentifyCmd identify;
+    NVMeRWCmd rw;
+    NVMeCreateCQCmd create_cq;
+    NVMeCreateSQCmd create_sq;
+};
```

Comment:
This should be a struct with fields `op`, `flags`, `cmdid`, (assuming these are shared fields) and then this union instead, this deduplicates the definitions as well as allowing future code to decide which of the union members should be accessed based on these shared fields.

Reply:
That's a good idea. Fixed it now.